service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RespondentControllerTest.java

```diff
@@ -90,6 +83,23 @@ void shouldReturnNoDateOfBirthErrorsForRespondentWhenValidDateOfBirth() {
         assertThat(callbackResponse.getErrors()).isEmpty();
     }
 
+    @Test
+    void shouldSetPersistRepresentativesFlagToYesOnRespondents() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("respondents1", buildRespondents()))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse 
+            = postMidEvent(caseDetails, "persist-representatives");
+
+        CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+        Respondent firstRespondent = caseData.getRespondents1().get(0).getValue();
+        Respondent secondRespondent = caseData.getRespondents1().get(0).getValue();
+
+        assertThat(firstRespondent.getPersistRepresentedBy()).isEqualTo(YES.getValue());
```

Comment:
Minor question - do you think it would be worth checking in above about to start or something that this value is not actually YES?

Reply:
Sorry, do you mean before we call the mid event (not about to start)?